plugins/IPBFormatter/class.ipbformatter.plugin.php

```diff
@@ -29,9 +29,9 @@ public function container_init(ContainerInterface $dic) {
         $formatService = $dic->get(FormatService::class);
         $formatService->registerFormat(IPBFormat::FORMAT_KEY, $dic->get(IPBFormat::class));
```

Comment:
I'd recommend moving these to the bootstrap file as well. `container_init` could cause some race conditions if something obtains a copy of the format service during plugin initialization.

Moving it to bootstrap ensures this race condition doesn't exist.

Reply:
You want to configure the container, not use it. You want something like this:

```php
$dic->rule(FormatService::class)
    ->addCall('registerFormat', [IPBFormat::FORMAT_KEY, new Reference(IPBFormat::class)])
```

This configures the container lazily so that it registers the IPB format when the format service is created.